presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/BuiltInQueryPreparer.java

```diff
@@ -0,0 +1,167 @@
+                            .map(PrestoWarning::getMessage)
+                            .collect(joining(System.lineSeparator()))));
+        }
+        return prepareQuery(analyzerOptions, wrappedStatement, preparedStatements);
+    }
+
+    public BuiltInPreparedQuery prepareQuery(AnalyzerOptions analyzerOptions, Statement wrappedStatement, Map<String, String> preparedStatements)
+            throws ParsingException, PrestoException, SemanticException
+    {
+        Statement statement = wrappedStatement;
+        Optional<String> prepareSql = Optional.empty();
+        if (statement instanceof Execute) {
+            String preparedStatementName = ((Execute) statement).getName().getValue();
+            prepareSql = Optional.ofNullable(preparedStatements.get(preparedStatementName));
+            String query = prepareSql.orElseThrow(() -> new PrestoException(NOT_FOUND, "Prepared statement not found: " + preparedStatementName));
+            statement = sqlParser.createStatement(query, analyzerOptions.getParsingOptions());
+        }
+
+        if (statement instanceof Explain && ((Explain) statement).isAnalyze()) {
+            Statement innerStatement = ((Explain) statement).getStatement();
+            Optional<QueryType> innerQueryType = StatementUtils.getQueryType(innerStatement.getClass().getSimpleName());
+            if (!innerQueryType.isPresent() || innerQueryType.get() == QueryType.DATA_DEFINITION) {
+                throw new PrestoException(NOT_SUPPORTED, "EXPLAIN ANALYZE doesn't support statement type: " + innerStatement.getClass().getSimpleName());
+            }
+        }
+        List<Expression> parameters = ImmutableList.of();
+        if (wrappedStatement instanceof Execute) {
+            parameters = ((Execute) wrappedStatement).getParameters();
+        }
+        validateParameters(statement, parameters);
+        Optional<String> formattedQuery = Optional.empty();
+        if (analyzerOptions.isLogFormattedQueryEnabled()) {
+            formattedQuery = Optional.of(getFormattedQuery(statement, parameters));
+        }
+        return new BuiltInPreparedQuery(wrappedStatement, statement, parameters, formattedQuery, prepareSql);
+    }
+
+    private static String getFormattedQuery(Statement statement, List<Expression> parameters)
+    {
+        String formattedQuery = formatSql(
+                statement,
+                parameters.isEmpty() ? Optional.empty() : Optional.of(parameters));
+        return format("-- Formatted Query:%n%s", formattedQuery);
+    }
+
+    private static void validateParameters(Statement node, List<Expression> parameterValues)
+    {
+        int parameterCount = getParameterCount(node);
+        if (parameterValues.size() != parameterCount) {
+            throw new SemanticException(INVALID_PARAMETER_USAGE, node, "Incorrect number of parameters: expected %s but found %s", parameterCount, parameterValues.size());
+        }
+        for (Expression expression : parameterValues) {
```

Comment:
how many spaces do we use for an indent in this code base?

Reply:
We use this code style: https://github.com/airlift/codestyle